### ISSUE TITLE
jaws run

### ISSUE THREAD

**Issue by remicastaing** [jaws-stack/JAWS#26]
If I understand it right, launching `jaws run` in an api folder performs an unit test according to the date provided in `event.js`rather than running the lambda function.

IMO, `jaws run` should rather set up an http listener with the right endpoint.

With a little `express` app, there could be possible to both serve the static part out of the site folder (I would rather name it `client`) and set up all the api endpoints.

Any thoughts?

And last but not least: a great thanks for your work.

**Comment by ac360** [comment]
@remicastaing I don't know if  `jaws run` should be considered a unit test.  

All `jaws run` does (currently) is emulate the AWS Lambda environment and run your code through it.  It does this by passing in the params that AWS Lambda passes into your function (event,context) and recreating the `context.success()`, context.fail()`, context.done()` methods that AWS Lambda requires you to call when your code has completed its run.  

Your suggestion about setting up an http listener with an endpoint is interesting.  It's something I've been thinking about and I believe it's part of the greater "How do we version and define environments for this stuff" question I've been working on over the last few days.  As of right now, I don't think the express http auto-generation is the easiest solution.  It's already very easy to deploy Lambda functions and REST APIs via the API Gateway, so I'd rather allow people to simply deploy a test API that way.  It's also the real thing and that's where I prefer to do my testing.

Stay tuned.  I'm going to unveil some new stuff which will attempt to answer this!  And thank you for your support!

**Comment by remicastaing** [comment]
@ac360, thank you to taking the time to provide this long response.

You're probably right saying simply deploying a lambda function would be the easiest way. I'm accustomed to have local environment for development purpose, so I though about how to get one for the lambda+api+S3+dynamoDB stack.  It could also be possible setting up a local dynamoDB instance.

**Comment by ac360** [comment]
@remicastaing I'm moving this conversation to #35.  In that thread we'll settle on a way to handle multiple environments.
